engine/variableValue.cc

```diff
@@ -51,20 +54,25 @@ namespace minsky
       case constant:
       case parameter:
         m_idx=ValueVector::flowVars.size();
-        ValueVector::flowVars.resize
-          (ValueVector::flowVars.size()+numElements());
+        //ValueVector::flowVars.resize(ValueVector::flowVars.size()+numElements());
+        // For feature 47
+		if (index.empty()) ValueVector::flowVars.resize(ValueVector::flowVars.size()+numDenseElements());
+		else ValueVector::flowVars.resize(ValueVector::flowVars.size()+numSparseElements);
```

Comment:
Could be improved by have a dedicated method that returns the number actual finite elements. Could be implemented as:

return index.size()? index.size(): numDenseElements();

Reply:
Thanks, I will implement this method today.

Reply:
Done. The segmentation faults have been fixed with this, thank you very much for the suggestion. The code now also works with the throw on line ~135 of variableValue.cc reinstated.